src/shared/format-table.c

```diff
@@ -1184,6 +1184,17 @@ int table_add_many_internal(Table *t, TableDataType first_type, ...) {
                         goto check;
                 }
 
+                case TABLE_SET_JSON_FIELD_NAME: {
+                        const char *n = va_arg(ap, const char*);
+                        size_t idx;
+                        if (t->vertical)
+                                idx = TABLE_CELL_TO_INDEX(last_cell) / t->n_columns - 1;
```

Comment:
hmm, please add an assert() here that gates the underflow (i.e. if TABLE_CELL_TO_INDEX(last_cell) is <  t->n_columns)

i.e. it's fine to not handle the underflow, after all that is a programmer bug. but please catch it with an assert here.

Reply:
I also added another check, in the !vertical case, to make sure TABLE_SET_JSON_FIELD_NAME is only applied onto a header cell